src/components/SampleSlotEditModal/SampleSlotEditModal.tsx

```diff
@@ -297,7 +307,14 @@ const SampleSlotEditModal: React.FC<SampleSlotEditModalProps> = ({
           <IonGrid>
             <IonRow>
               <IonCol>
-                <IonButton fill="outline" expand="block" onClick={onCancel}>
+                <IonButton
+                  fill="outline"
+                  expand="block"
+                  // Call dismiss here instead of onCancel directly so that the modal can animate
+                  // out of view before the onCancel handler is called via the IonModalDidDismiss
+                  // event
```

Comment:
```suggestion
                  // Call `dismiss` here instead of `onCancel` directly so that the modal can animate
                  // out of view before the `onCancel` handler is called via the `IonModalDidDismiss`
                  // event.
```